readthedocs/v3/serializers.py

```diff
@@ -0,0 +1,445 @@
+                'parent_lookup_version__slug': obj.slug,
+            },
+        )
+        return self._absolute_url(path)
+
+    def get_project(self, obj):
+        path = reverse(
+            'projects-detail',
+            kwargs={
+                'project_slug': obj.project.slug,
+            },
+        )
+        return self._absolute_url(path)
+
+
+class VersionURLsSerializer(serializers.Serializer):
+    documentation = serializers.SerializerMethodField()
+    vcs = serializers.URLField(source='vcs_url')
+
+    def get_documentation(self, obj):
+        return obj.project.get_docs_url(version_slug=obj.slug,)
+
+
+class VersionSerializer(FlexFieldsModelSerializer):
+
+    privacy_level = PrivacyLevelSerializer(source='*')
+    ref = serializers.CharField()
+    downloads = serializers.SerializerMethodField()
+    urls = VersionURLsSerializer(source='*')
+    links = VersionLinksSerializer(source='*')
+
+    expandable_fields = dict(
+        last_build=(
+            BuildSerializer,
+            dict(
+                source='last_build',
+            ),
+        ),
+    )
+
+    class Meta:
+        model = Version
+        fields = [
+            'id',
+            'slug',
+            'verbose_name',
+            'identifier',
+            'ref',
+            'built',
+            'active',
+            'uploaded',
+            'privacy_level',
+            'type',
+            'downloads',
+            'urls',
+            'links',
+        ]
+
+    def get_downloads(self, obj):
+        downloads = obj.get_downloads()
+        data = {}
+
+        for k, v in downloads.items():
+            if k in ('htmlzip', 'pdf', 'epub'):
+                data[k] = ('http:' if settings.DEBUG else 'https:') + v
+
+        return data
+
+
+class VersionUpdateSerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = Version
+        fields = [
+            'active',
+            'privacy_level',
+        ]
+
+
+class LanguageSerializer(serializers.Serializer):
+
+    code = serializers.SerializerMethodField()
+    name = serializers.SerializerMethodField()
+
+    def get_code(self, language):
+        return language
+
+    def get_name(self, language):
+        for code, name in LANGUAGES:
+            if code == language:
+                return name
+        return 'Unknown'
+
+
+class ProgrammingLanguageSerializer(serializers.Serializer):
+
+    code = serializers.SerializerMethodField()
+    name = serializers.SerializerMethodField()
+
+    def get_code(self, programming_language):
+        return programming_language
+
+    def get_name(self, programming_language):
+        for code, name in PROGRAMMING_LANGUAGES:
+            if code == programming_language:
+                return name
+        return 'Unknown'
+
+
+class ProjectURLsSerializer(serializers.Serializer):
+    documentation = serializers.CharField(source='get_docs_url')
+    project = serializers.SerializerMethodField()
```

Comment:
Looks at this in the response was confusing:

```
urls: {
documentation: "http://time-test.dev.readthedocs.io:8000/en/latest/",
project: null
}
```

Perhaps it should be `project_homepage` or something? Otherwise I might confuse it with the URL of the RTD page for the project (which isn't in the response, afaict)

Reply:
Mmm... good point. I didn't add "human" URLs for Project, Version and Build under Read the Docs. I may worth to add them all in each response.